File: src/models/rate_limit_token_bucket_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct RateLimitTokenBucketConfig {
    /// Maximum capacity of the bucket
    pub capacity: u64,

    /// Number of tokens to add per refill interval
    pub refill_amount: u64,

    /// Interval in milliseconds between refills (minimum 1 millisecond)
    #[serde(
        rename = "refill_interval_ms",
        skip_serializing_if = "Option::is_none",
        with = "crate::duration_ms_serde::optional"
    )]
    pub refill_interval: Option<std::time::Duration>,
}

impl RateLimitTokenBucketConfig {
    pub fn new(capacity: u64, refill_amount: u64) -> Self {
        Self {
            capacity,
            refill_amount,
            refill_interval: None,
        }
    }

    pub fn with_refill_interval(mut self, value: impl Into<Option<std::time::Duration>>) -> Self {
        self.refill_interval = value.into();
        self
    }
}

File: coyote/models/rate_limit_token_bucket_config.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class RateLimitTokenBucketConfig(BaseModel):
    capacity: int
    """Maximum capacity of the bucket"""

    refill_amount: int = Field(alias="refill_amount")
    """Number of tokens to add per refill interval"""

    refill_interval_ms: t.Optional[int] = Field(
        default=None, alias="refill_interval_ms"
    )
    """Interval in milliseconds between refills (minimum 1 millisecond)"""
